features/bsm/service/api/src/main/java/org/opennms/netmgt/bsm/service/BusinessServiceSearchCriteriaBuilder.java

```diff
@@ -0,0 +1,318 @@
+     *
+     * @param <A> an object
+     * @param <B> another object
+     */
+    private static class Pair<A, B> {
+        public final A a;
+        public final B b;
+
+        /**
+         * Constructor for instatiating this class.
+         *
+         * @param a one object
+         * @param b another object
+         */
+        public Pair(A a, B b) {
+            this.a = a;
+            this.b = b;
+        }
+
+        public String toString() {
+            return "Pair (" + this.a + "," + this.b + ")";
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Pair<?, ?> pair = (Pair<?, ?>) o;
+
+            if (a != null ? !a.equals(pair.a) : pair.a != null) {
+                return false;
+            }
+            return b != null ? b.equals(pair.b) : pair.b == null;
+
+        }
+
+        @Override
+        public int hashCode() {
+            int result = a != null ? a.hashCode() : 0;
+            result = 31 * result + (b != null ? b.hashCode() : 0);
+            return result;
+        }
+
+        /**
+         * Returns the first associated object.
+         *
+         * @return the object
+         */
+        public A getA() {
+            return a;
+        }
+
+        /**
+         * Returns the second associated object.
+         *
+         * @return the object
+         */
+        public B getB() {
+            return b;
+        }
+
+        /**
+         * Constructs an instance for two given objects.
+         *
+         * @param a   the object
+         * @param b   another object
+         * @param <A> the type of the first object
+         * @param <B> the type of the second object
+         * @return the instance created
+         */
+        public static <A, B> Pair<A, B> of(A a, B b) {
+            return new Pair(a, b);
+        }
+    }
+
+    /**
+     * the name filters
+     */
+    private List<String> m_nameFilters = new ArrayList<>();
+    /**
+     * the severity filters
+     */
+    private List<Pair<CompareOperator, OnmsSeverity>> m_severityFilters = new ArrayList<>();
+    /**
+     * the attribute filters
+     */
+    private List<Pair<String, String>> m_attributeFilters = new ArrayList<>();
+    /**
+     * the order parameters
+     */
+    private Order m_order = Order.Name;
+    /**
+     * the results limit
+     */
+    private int m_limit = 0;
+    /**
+     * ascending or descending?
+     */
+    private Sequence m_sequence = Sequence.Ascending;
+
+    /**
+     * Default constructor
+     */
+    public BusinessServiceSearchCriteriaBuilder() {
+    }
+
+    @Override
+    public List<BusinessService> apply(BusinessServiceManager businessServiceManager, List<BusinessService> businessServiceDTOs) {
+        Stream<BusinessService> s = businessServiceDTOs.stream();
```

Comment:
We could do that, but then we cannot filter for the severity/status of the Business Services as we do not persist it. I'd say let's not worry about it at the moment as we have to come back to address this later anyways.